### PR TITLE
Workflow changes

### PR DESCRIPTION
**Pull request by rajpalc7**

### CODE REVIEW

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
I think you need to rebase your updates to these docs on the latest version in the upstream `bcgov/DITP-DevOps` project.  This PR appears to be duplicating changes you made in https://github.com/bcgov/DITP-DevOps/pull/35.  We'll run into merge conflicts tryin the merge these changes.  Most of these files already exist in the repo, but your PR is trying to add them as new files.

File: docs/bc-registries-audit-notification-workflow.md [hunk 1]
```diff
@@ -88,4 +88,15 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
-This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
\ No newline at end of file
+This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
+
+## Troubleshoot Failed Credentials Error
+
+If there are companies that dont't get re-processed because the credentials were created in the credential_log table in the event processor database. Follow the steps below to troubleshoot failed credentials error:
+
+Run the `./manage -e prod runPipeline`command and verify the credentials errors. An example regarding this is shown below in the image:
+[BC-Registries-Audit-Errors](./images/bc-registries-audit-errors.png)
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:98]: `[BC-Registries-Audit-Errors](./images/bc-registries-audit-errors.png)`, The image isn't linked properly for it to be displayed.  Use the other image links in the document as a reference.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Way better.  Good work.  One small issue below.

File: docs/bc-registries-audit-notification-workflow.md (hunk 1 above)
Comment (rajpalc7) [reply docs/bc-registries-audit-notification-workflow.md]: Thanks Wade, I have fixed the image link now.

File: docs/bc-registries-audit-notification-workflow.md [hunk 2]
```diff
@@ -88,4 +88,17 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
 
-This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
\ No newline at end of file
+This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
+
+## Troubleshoot Failed Credentials Error
+
+If there are companies that dont't get re-processed because the credentials were created in the credential_log table in the event processor database. Follow the steps below to troubleshoot failed credentials error:
+
+Run the `./manage -e prod runPipeline`command and verify the credentials errors. An example regarding this is shown below in the image:
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:97]: For consistency and ease of reuse, please update `bc-registries-agent-configurations-manage-script.md`.  Add a section for `./manage -e prod runPipeline` describing it's use and link to it from here.  i.e. Follow the same pattern used with the `requeueOrganization` and `queueOrganization` commands.

File: docs/bc-registries-audit-notification-workflow.md [hunk 3]
```diff
@@ -88,4 +88,17 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
+
+![BC-Registries-Audit-Errors](./images/bc-registries-audit-errors.png)
+
+
+Running the `./manage -e prod requeueFailedCreds` command will clear and solve the status of Failed Credentials.
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:102]: For consistency and ease of reuse, please update `bc-registries-agent-configurations-manage-script.md`.  Add a section for `./manage -e prod requeueFailedCreds` describing it's use and link to it from here.  i.e. Follow the same pattern used with the `requeueOrganization` and `queueOrganization` commands.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
A couple improvements to make.

**Comment by rajpalc7** [comment]
@WadeBarnes - Added a section for ./manage -e prod requeueFailedCreds describing it's use and linked

File: docs/bc-registries-audit-notification-workflow.md [hunk 4]
```diff
@@ -88,4 +88,17 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
 
-This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
\ No newline at end of file
+This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
+
+## Troubleshoot Failed Credentials Error
+
+If there are companies that dont't get re-processed because the credentials were created in the credential_log table in the event processor database. Follow the steps below to troubleshoot failed credentials error:
+
+Run the `./manage -e prod runPipeline` command and verify the credentials errors. An example regarding this is shown below in the image:
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:97]:
```suggestion
Run the `./manage -e prod getPipelineStatus` command and verify the credentials errors. An example regarding this is shown below in the image:
```

Sorry I missed this the first time.

For consistency and ease of reuse, please update `bc-registries-agent-configurations-manage-script.md`.  Add a section for `./manage -e prod getPipelineStatus` describing it's use and link to it from here.  i.e. Follow the same pattern used with the `requeueOrganization`, `queueOrganization`, and `requeueFailedCreds` commands.

The `./manage -e prod runPipeline` command is what you run at the end after you have requeued any records using the other commands.  You should add this to the documentation and make a section for it in the `bc-registries-agent-configurations-manage-script.md` as well so you can link to it.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]

**Comment by rajpalc7** [comment]
@WadeBarnes - Added a section for run pipeline describing it's use and linked

File: docs/bc-registries-audit-notification-workflow.md [hunk 5]
```diff
@@ -88,4 +88,19 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
 
-This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
\ No newline at end of file
+This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
+
+## Troubleshoot Failed Credentials Error
+
+If there are companies that dont't get re-processed because the credentials were created in the credential_log table in the event processor database. Follow the steps below to troubleshoot failed credentials error:
+
+Run the `./manage -e prod getPipelineStatus` command and verify the credentials errors. An example regarding this is shown below in the image:
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:97]: For consistency and ease of reuse, please update `bc-registries-agent-configurations-manage-script.md`.  Add a section for `./manage -e prod getPipelineStatus` describing its use and link to it from here.  i.e. Follow the same pattern used with the `requeueOrganization`, `queueOrganization`, and `requeueFailedCreds` commands.

You will find a description for this command here; [getPipelineStatus command documentation](https://github.com/bcgov/von-bc-registries-agent-configurations/blob/main/openshift/manage#L92-L95)

File: docs/bc-registries-audit-notification-workflow.md [hunk 6]
```diff
@@ -88,4 +88,19 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
+
+![BC-Registries-Audit-Errors](./images/bc-registries-audit-errors.png)
+
+
+Running the [`requeueFailedCreds` command](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command will clear and solve the status of Failed Credentials.
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:102]:
```suggestion
From the output above you can see there are 50 processing errors in the `credential_log` table.  This means there are 50 credentials that have not been processed and posted to the OrgBook.  To requeue these credentials for processing run the [`requeueFailedCreds`](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command.
```

File: docs/bc-registries-audit-notification-workflow.md [hunk 7]
```diff
@@ -88,4 +88,19 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
+
+Running the [`requeueOrganization` command](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) and [`queueOrganization` command](./bc-registries-agent-configurations-manage-script.md#queueOrganization-command) will resolve the data issue.
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:104]: I don't think this paragraph is relevant to this particular example, is it?  If not please delete.

File: docs/bc-registries-audit-notification-workflow.md [hunk 8]
```diff
@@ -88,4 +88,19 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
+
+Run the [`runPipeline` command](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command at the end after all the records are requeued using the other commands.
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:106]:
```suggestion
Once you've verified the credentials have been requeued, run the [`runPipeline` ](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command to process the outstanding records.
```

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Additional commands and recommendations below.

File: docs/bc-registries-agent-configurations-manage-script.md [hunk 9]
```diff
@@ -56,4 +56,132 @@ Executing command on 7cba16-dev/event-db-primary-20-9x4rc:
...
+Executing command on 7cba16-prod/event-db-primary-8-q6p2q:
+        psql -d ${POSTGRESQL_DATABASE} -ac "update credential_log set process_success = null, process_date = null, process_msg = null where process_success = 'N';"
+
+update credential_log set process_success = null, process_date = null, process_msg = null where process_success = 'N';
+UPDATE 0
+```
+
+## `getPipelineStatus` command
+
+This command will verify the credentials errors that has processed and provide current status of Pipeline.
```
Comment (WadeBarnes) [inline docs/bc-registries-agent-configurations-manage-script.md:82]:
```suggestion
This command gets the pipeline status for the given environment.  It runs the `./run-step.sh bcreg/display_pipeline_status.py` pipeline script on an instance of the event-processor pod.
```

The content I'm suggesting here came from the `getPipelineStatus` command description I provided a link to in my previous review comments; [getPipelineStatus command documentation](https://github.com/bcgov/von-bc-registries-agent-configurations/blob/main/openshift/manage#L92-L95)

File: docs/bc-registries-audit-notification-workflow.md [hunk 10]
```diff
@@ -88,4 +88,12 @@ Specifics regarding the commands referenced in this workflow can be found here:
...
\ No newline at end of file
+This repository provides scripts to audit the OrgBook search database and agent wallet against the source BC Registries data; [Audit Scripts](https://github.com/bcgov/von-bc-registries-audit/blob/main/README.md#understanding-the-output)
+
+## Troubleshoot Failed Credentials Error
+
+Occasionally credentials get stuck in the processing queue due to errors, and will not respond to the use of the [`requeueOrganization`](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) and/or [`queueOrganization`](./bc-registries-agent-configurations-manage-script.md#queueOrganization-command) commands.  In these cases you want to check the pipeline status for an errors.
+
+Running the [`getPipelineStatus` command](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command will verify the credentials errors.
+
+After verifying the credentials have been requeued, run the [`runPipeline` ](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) command to process the outstanding records.
```
Comment (WadeBarnes) [inline docs/bc-registries-audit-notification-workflow.md:99]: You removed too much content from this section and did not add the other recommendations I made.  Also the screenshot from `getPipelineStatus` got removed from the PR.

I see there may have been some confusion regarding this comment; https://github.com/bcgov/DITP-DevOps/pull/54#discussion_r1114383899

Is was referencing this paragraph only, all other content should have stayed:

> Running the [`requeueOrganization` command](./bc-registries-agent-configurations-manage-script.md#requeueOrganization-command) and [`queueOrganization` command](./bc-registries-agent-configurations-manage-script.md#queueOrganization-command) will resolve the data issue.

Comment (rajpalc7) [reply docs/bc-registries-audit-notification-workflow.md]: Hi @WadeBarnes made the changes according to your suggestion

**Review by WadeBarnes — APPROVED** [review]
Nice work!

**Pull request merged**